README.md

```diff
@@ -128,15 +165,36 @@ OakMap<K,V> oak = builder.build();
-Note, however, that since OakMap's get method avoids copying the value and instead returns access to the same underlying memory buffer that compute operations update in-place, the reader may encounter different values -- and even value deletions -- when accessing the buffer returned from get multiple times. This is of course normal behavior for a concurrent map that avoids copying.
-
-An OakRBuffer can represent either a key or a value. The OakRBuffer's user can use the standard interface of a *read-only* ByteBuffer, for example, `int getInt(int index)`, `char getChar(int index)`, `limit()`, etc. Note that ConcurrentModificationException can be thrown as a result of any OakRBuffer method in case the mapping is concurrently deleted.
-
-For backward compatibility with applications that are already based on the use of ByteBuffers, Oak Buffers provide the transform method that atomically applies a transformation to a *read-only* instance of the underlying ByteBuffer. For a more comprehensive code example please refer to the [usage](#usage) section. 
+### Oak Buffers
+Oak use dedicated buffer objects to access off-heap memory. 
+These buffers cannot be instantiated by the user and are always supplied to the user by Oak.
+Their interfaces are:
+ 1. `OakReadBuffer`: scoped, read-only access to the data
+ 2. `OakWriteBuffer`: scoped, read and write access to the data
+ 3. `OakScopeFreeBuffer`: scope-free, read-only access to the data
+
+These buffers may represent either a key or a value.
+They mimic the standard interface of Java's `ByteBuffer`, for example, `int getInt(int index)`, `char getChar(int index)`, `capacity()`, etc. 
+
+Scoped buffers (`OakReadBuffer` and `OakWriteBuffer`) can only be used in the scope of the callback method they were first introduced to the user. 
+Such callback method might be the application's serializer and comparator, or a lambda function that can read/store/update the data (see [full API](https://github.com/yahoo/Oak/wiki/Full-API) for more information).
+This access reduces unnecessary copies and deserialization of the underlying data.
+In their intended context, the user does not need to worry about concurrent accesses and memory management.
+Using these buffers outside their intended context may yield unpredicted results, e.g., reading non-consistent data and/or irrelevant data.
+
+The third buffer (`OakScopeFreeBuffer`) is scope free, i.e., its usage is not scope limited and may be stored for future use.
+Some OakMap's methods returns this buffer to avoid copying the data and instead the user can access to the underlying memory buffer directly (lazy evaluation).
+While the scoped buffers are mutually exclusive, when using `OakScopeFreeBuffer`, the same memory might be access by a concurrent update operations.
```

Comment:
"mutually exclusive" is not clear enough, explain that this is about synchronization of the data access/update. In addition multiple read accesses can happen simultaneously, so not exactly mutually exclusive. May be just say: "data accesses are synchronized..."

Reply:
fixed